test/metabase/domain_entities/core_test.clj

```diff
@@ -0,0 +1,46 @@
+(ns metabase.domain-entities.core-test
+  (:require [expectations :refer [expect]]
+            [metabase.domain-entities.core :as de]
+            [metabase.models
+             [field :refer [Field]]
+             [table :as table :refer [Table]]]
+            [metabase.test
+             [data :as data]
+             [domain-entities :refer :all]]))
+
+
+(expect
+  [:field-id (data/id :venues :price)]
+  (#'de/mbql-reference (Field (data/id :venues :price))))
+
+(expect
+  [:field-literal "PRICE" :type/Integer]
+  (#'de/mbql-reference (dissoc (Field (data/id :venues :price)) :id)))
+
+
+(defn- hydrated-table
+  [table-name]
+  (let [table (-> table-name data/id Table)]
+    (assoc table :fields (table/fields table))))
```

Comment:
you can just do `(toucan/hydrate table :fields)`

Reply:
bump

Reply:
stale